src/foam/core/XMLSupport.java

```diff
@@ -0,0 +1,199 @@
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.FileWriter;
+import com.sun.xml.internal.txw2.output.IndentingXMLStreamWriter;
+import foam.dao.XMLDAO;
+import foam.dao.ListSink;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.ZoneId;
+
+public class XMLSupport {
+
+  // TODO: Planning to add support for other date formats in the future
+
+  public static List<FObject> fromXML(XMLStreamReader xmlr) {
+    List<FObject> objList = new ArrayList<FObject>();
+    try {
+      int eventType;
+      Object clsInstance = null;
+      while ( xmlr.hasNext() ) {
+        eventType = xmlr.next();
+        switch ( eventType ) {
+          case XMLStreamConstants.START_DOCUMENT:
+            break;
+          case XMLStreamConstants.START_ELEMENT:
+            if (xmlr.getLocalName().equals("object")) {
+              // Create new fObject
+              String objClass = xmlr.getAttributeValue(null, "class");
+              Class<?> cls;
+
+              try {
+                cls = Class.forName(objClass);
+                clsInstance = cls.newInstance();
+              } catch (ClassNotFoundException | InstantiationException ex) {
+
+              }
+              // Object properties
+              copyFromXML((FObject) clsInstance, xmlr);
+              if (clsInstance != null) {
+                objList.add((FObject) clsInstance);
+              }
+            }
+            break;
+          case XMLStreamConstants.END_ELEMENT:
+            break;
+          case XMLStreamConstants.END_DOCUMENT:
+            break;
+        }
+      }
+      xmlr.close();
+    } catch (XMLStreamException | IllegalAccessException ex) {
+    }
+    return objList;
+  }
+
+  public static List<FObject> fromXML(String fileName) throws IOException {
+    XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+    XMLStreamReader xmlr = null;
+    try {
+      xmlr = xmlInputFactory.createXMLStreamReader(new FileReader(fileName));
+    } catch (IOException ex) {
+      throw new IOException("Could not create/file with given fileName");
+    } catch (XMLStreamException ex) {
+
+    }
+    return fromXML(xmlr);
+  }
+
+  public static void copyFromXML(FObject obj, XMLStreamReader reader) throws XMLStreamException {
+    try {
+      PropertyInfo prop = null;
+      while ( reader.hasNext() ) {
+        int eventType;
+        eventType = reader.next();
+        switch ( eventType ) {
+          case XMLStreamConstants.START_ELEMENT:
+            prop = (PropertyInfo) obj.getClassInfo().getAxiomByName(reader.getLocalName());
+            if ( prop == null ) {
+              throw new XMLStreamException("Could not find property " + reader.getLocalName());
```

Comment:
Just ignore, or at most log a warning. Not an error to have extra data. This will make it difficult to migrate data.